modules/node_modules/@frogpond/ccci-stolaf-college/v1/jobs/index.js

```diff
@@ -1,52 +1,85 @@
 import {get, ONE_DAY} from '@frogpond/ccc-lib'
 import mem from 'mem'
+import pMap from 'p-map'
 import _jsdom from 'jsdom'
 import getUrls from 'get-urls'
 
 const {JSDOM} = _jsdom
 
-export function cleanJob(job) {
-	// these all need to retain their newlines
-	const description = cleanTextBlock(
-		JSDOM.fragment(job.description).textContent,
-	)
-	const comments = cleanTextBlock(JSDOM.fragment(job.comments).textContent)
-	const skills = cleanTextBlock(JSDOM.fragment(job.skills).textContent)
-	const howToApply = cleanTextBlock(JSDOM.fragment(job.howToApply).textContent)
-	const timeline = cleanTextBlock(JSDOM.fragment(job.timeline).textContent)
-	const timeOfHours = cleanTextBlock(
-		JSDOM.fragment(job.timeOfHours).textContent,
-	)
-
-	const contactEmail = fixupEmailFormat(job.contactEmail)
-	const contactPhone = fixupPhoneFormat(job.contactPhone)
+const jobsUrl =
+	'https://wp.stolaf.edu/student-jobs/wp-json/wp/v2/pages/80'
+
+const PARAGRAPHICAL_KEYS = ['Job Description', 'Skills Needed', 'Additional Comments', 'How to Apply', 'Hiring Timeline']
+
+async function fetchDetail(url) {
+	const resp = await get(url)
+
+	// run-scripts value is needed to properly evaluate javascript to display an email address.
+	// see the jsdom documentation for more details https://github.com/jsdom/jsdom#executing-scripts
+	const dom = new JSDOM(resp.body, {contentType: 'text/html', runScripts: 'dangerously' })
+
+	const id = url.replace(/\D/g,'')
+	const titleText = dom.window.document.querySelector('.gv-list-view-title > h3').textContent
+	const details = dom.window.document.querySelectorAll('div')
+
+	const detailMap = [...details].reduce((coll, listEl) => {
```

Comment:
Can you add a comment describing what this is all doing to the DOM structure? Maybe include some example structure as we receive it and then highlight what transformations are taking place? It's hard for me to tell without tests or comments what the expected input for this function is, so in the future if the request format changes it's not likely to be super fun to figure out what needs to change.

Reply:
Sure, I can do that. I took the implementation from the Carleton jobs html parsing. We could probably make this a reusable function across the modules. https://github.com/frog-pond/ccc-server/blob/master/modules/node_modules/@frogpond/ccci-carleton-college/v1/jobs/index.js